pvanalytics/features/shading.py

```diff
@@ -0,0 +1,338 @@
+"""Functions for labeling shading and shodows."""
+import numpy as np
+import pandas as pd
+from scipy import ndimage
+from skimage import morphology, measure
+import pvlib
+from pvanalytics import util
+
+
+def _to_image(data, width):
+    """Convert data to an image.
+
+    Parameters
+    ----------
+    data : Series
+        Values of the pixels
+    width : int
+        Width of the image in pixels
+
+    Returns
+    -------
+    ndarray
+        Data reshaped into an ndarray with width `width`
+    """
+    return data.reshape(len(data) // width, width)
+
+
+def _to_series(image, image_index):
+    """Convert a binary image to a boolean series.
+
+    Parameters
+    ----------
+    image : ndarray
+        Binary image.
+    image_index : pd.Index
+        Index for the returned series.
+
+    Returns
+    -------
+    Series
+        A pandas Series with index `index` and values from the image.
+    """
+    return pd.Series(image.flatten(), index=image_index)
+
+
+def _smooth(image):
+    """Smooth `image` by local mean filtering.
+
+    Local means are computed in a 3x3 square surrounding each pixel.
+    """
+    return ndimage.uniform_filter(image, size=15)
+
+
+def _prepare_images(ghi, clearsky, daytime, interval):
+    """Prepare data as images.
+
+    Performs pre-processing steps on `ghi` and `clearsky` before
+    returning images for use in the shadow detection algorithm.
+
```

Comment:
Add brief description of input parameters. It is clear in the usage that e.g. `ghi` and `clearsky` are `Series` with the same index, and `interval` is in minutes